src/keylessBackup/encryption.ts

```diff
@@ -0,0 +1,91 @@
+import crypto from 'crypto'
+import { ec as EC } from 'elliptic'
+import hkdf from 'futoin-hkdf'
+
+const ec = new EC('secp256k1')
+
+/**
+ * Derives a 256-bit key using the HKDF method from two key shares.
+ *
+ * @param {string} keyshare1 - The first keyshare in utf8 format.
+ * @param {string} keyshare2 - The second keyshare in utf8 format.
+ * @returns {Buffer} A derived 256-bit key as a Buffer.
+ */
+export function deriveKeyFromKeyShares(keyshare1: string, keyshare2: string): Buffer {
+  // Combining the keyshares
+  const combinedKeyShares = Buffer.concat([
+    Buffer.from(keyshare1, 'utf8'),
+    Buffer.from(keyshare2, 'utf8'),
+  ])
+
+  // Using futoin-hkdf to derive a 256-bit key
+  return hkdf(combinedKeyShares, 32, {
+    salt: 'some fixed salt',
+    info: 'encryption',
+    hash: 'SHA-256',
+  })
+}
+
+/**
+ * Derives a 256-bit key using the HKDF method from two key shares then derives a secp256k1 pair from that.
+ *
+ * @param {string} keyshare1 - The first keyshare in utf8 format.
+ * @param {string} keyshare2 - The second keyshare in utf8 format.
+ * @returns {Buffer} A derived 256-bit key as a Buffer.
+ */
+export function getSecp256K1KeyPair(keyshare1: string, keyshare2: string): EC.KeyPair {
+  const derivedKey = deriveKeyFromKeyShares(keyshare1, keyshare2)
+
+  return ec.keyFromPrivate(derivedKey)
```

Comment:
This always returns undefined in unit tests, so I believe this to be incorrect. I believe its because you can't just use any odd HKDF encryption key as a derivedKey for ec.keyFromPrivate, but I'm not sure how else I am supposed to get a secp256k1 key from the combined keyshare.

Reply:
If you do switch to noble secp256k1, you can use the `hashToPrivateKey: (hash: Hex)` [1] function. Looking at it's implementation, it accepts a hex string of at least 40 bytes, and produces a private key for you (Note that the "at least 40 bytes" is required to avoid bias in the resulting key, which would somewhat reduce it's strength).

I would recommend passing in 48 bytes simply because it's 32+16, which means the bias in the resulting key will be less that 2^-128, which is the gold standard for cryptography implementation.

[1] https://github.com/paulmillr/noble-secp256k1#utils